### ISSUE TITLE
[Enhancement]: Combine eBooks and Audiobooks from different folders in the same library

### ISSUE THREAD

**Issue by mew1033** [advplyr/audiobookshelf#3062]
### Type of Enhancement

Web Interface/Frontend

### Describe the Feature/Enhancement

I would love it if I could point to two separate folders for my library, one with audiobooks and one with ebooks, then have those folders combine into the same item in the frontend. I don't want to store my eBooks and Audiobooks in the same folder, but I _do_ want them to appear together in audiobookshelf. For now, I solve this by adding symlinks on my NAS so the eBooks appear in the Audiobook's folder. But it'd be awesome if this could be supported natively.

The major issue here is matching books together. What if some of my books only have eBooks? What if they get matched incorrectly? Perhaps there could be a metadata field that links them together? Or a matching interface where you help audiobookshelf know that two separate items are actually the same item.

### Why would this be helpful?

I want to store my eBooks and Audiobooks in different folders, but have them appear as a combined item in audiobookshelf.

### Future Implementation (Screenshot)

![image](https://github.com/advplyr/audiobookshelf/assets/4435733/513f25ee-70b5-440e-a16b-ba1e324d4069)
![image](https://github.com/advplyr/audiobookshelf/assets/4435733/c60c9ddf-8629-44b2-abc8-995d87623225)

### Audiobookshelf Server Version

v2.10.1

**Comment by nichwall** [comment]
Duplicate of https://github.com/advplyr/audiobookshelf/issues/2588

**Comment by mew1033** [comment]
Man, I looked forever and couldn't find a duplicate. Sorry about that! Happy to close this in favor of that. Or if you want the new issue template used, this can stay open.

**Comment by advplyr** [comment]
We'll keep this one open. 

Related to this is putting multiple books with different narrators (or qualities) on the same book page. https://github.com/advplyr/audiobookshelf/issues/2396

Abs data model has a page for each library item (book or podcast) and each library item has a file path so the challenge with this is similar to #2396.

**Comment by sthompson-celerity** [comment]
I know this is not super helpful to everyone, but for those using linux this is a very viable solution, I am currently testing mergefs to achieve this functionality. Hopefully everything goes well...

**Comment by dankozlowski** [comment]
I'd love to help with this, if someone is currently working on it.

**Comment by dankozlowski** [comment]
@sthompson-celerity I'd love to help out with this.  Let me know if there's any way I can assist.

This would be an amazing improvement to ABS.

**Comment by fivestones** [comment]
> What would be nice is being able to:
> 
> 1. Identify distinct editions/adaptations/abridgments/translations/narrations of the same original 'work' and link them together
> 2. Collapse them or expand them in the list view just like Series
> 3. See the linked entries while on the detailed view
> 4. Star/Favorite or otherwise mark preferred versions of a work (or optimally give them a numeric order of preference)

(quoting @manalive from https://github.com/advplyr/audiobookshelf/issues/2396)

I think this would be ideal. If you open the detailed view for any one book version (audiobook edition/ebook edition/adaptation/etc) of a given original 'work', you see the one you opened as it is now, but below this you see a list of all the other editions/versions of the same work. Preferably you would not have to click again to open those book editions before reading/listening, but could read/listen to a linked edition directly from that same page. Maybe something like the image below.

![audiobookshelf linked books](https://github.com/user-attachments/assets/5540bf4b-9e0f-4ea5-990f-c164a570cd22)

**Comment by mikemathewson** [comment]
This is even more necessary when using Readarr. With Readarr if you want to download both audiobooks and ebooks you must install separate instances and each instance must have a different root folder.

**Comment by enieuwy** [comment]
Yes, this my conundrum. I’m not ready to ditch my two readarr instances to combine the libraries, but I really really want the assets combined in ABS!
________________________________

**Comment by sicahjes** [comment]
If it helps, I would love to see this options but for a different use case. I will be setting this up so that I can share ebooks and audiobooks with family, all of whom have varying degrees of tech ability and vastly different preferences. Some are only interested in ebooks, others audiobooks, and some have a use for both. 

- If I had a mixed library, some of my users (the more elderly) will get confused and overwhelmed by the duplication. I can see them giving up when they try to access an ebook and it's an audiobook etc. 
- At this stage I think I'll have three libraries: audiobooks, non-fiction ebooks, and fiction e-books. 
- It would be wonderful to be able to open an ebook (for example) and see that there is also an audiobook available without having to use double the space for audiobooks (by duplicating the files in the ebook libraries) or mixing everything. 

I don't know if linked books is a possibility (referring to the above comment) but I love the idea of a linked field that works as a recommendation list based in shared attributes.  

_Full disclosure: I haven't set this up yet, it will be my first foray into self hosting and I am in the heavy 'research' phase (so apologies if there's something I've missed)._

**Comment by sthompson-celerity** [comment]
> @sthompson-celerity I'd love to help out with this.  Let me know if there's any way I can assist.
> 
> This would be an amazing improvement to ABS. 

Sorry, I kinda just forgot about this thread. It worked great. merge fs handled everything. I switched to using lazy librarian though, and it handles this without having to use mergefs. lazy librarian is WAY better for books than any other service i tested. You can download audio books and epubs to the same folder with lazy librarian, so this setup is not even neccessary.

**Comment by enieuwy** [comment]
Would love to know more about how you did that?

________________________________

**Comment by sthompson-celerity** [comment]
It was not hard, I made a new folder that was a combination of the book library, and the audio library that has matching names so the authors and books folders all had the exact same name, then I ran the command in mergefs to combine the two into the new folder. The only problem was that the obfs and the audiobook shelf json comes were overwriting easier, but that settled out after a bit and I think I choose an option in mergefs to only use the newest one I think

**Comment by enieuwy** [comment]
Thanks for responding. Clever—but matching the folder names isn't easy for me given Calibre's folder numbering.

**Comment by Offbeatmammal** [comment]
I've shared the solution I build for our library at home - simple PHP site that runs on the same server and allows you to upload an ebook to the existing folders (my installation is via Docker, but the library lives outside) https://github.com/Offbeatmammal/abs-plus-ebook
